lib/src/ruby/module.cc

```diff
@@ -908,16 +918,19 @@ namespace facter { namespace ruby {
         return it->second;
     }
 
-    VALUE module::execute_command(std::string const& command, VALUE failure_default, bool raise, uint32_t timeout)
+    VALUE module::execute_command(std::string const& command, VALUE failure_default, bool raise, uint32_t timeout, bool expand)
     {
         auto const& ruby = api::instance();
-
-        // Expand the command
-        auto expanded = expand_command(command);
-
+        std::string expanded;
+        try {
+        // Expand the command only if expand is true,
+        expanded = leatherman::execution::expand_command(command, leatherman::util::environment::search_paths(), expand);
```

Comment:
leatherman::execution is already imported, no need to specify it explicitly here and in similar places